modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java

```diff
@@ -297,9 +297,9 @@ public void processOpts() {
             LOGGER.info("Set base package to invoker package ({})", basePackage);
         }
 
-        super.processOpts();
         useOneOfInterfaces = true;
         legacyDiscriminatorBehavior = false;
+        super.processOpts();
```

Comment:
You may want to leave a comment here since where this is called matters; otherwise someone could try to move it again later (even though you have tests for it)

Reply:
I like that suggestion, it could be helpful for unrelated changes in the future as well :)